src/tests/eet/eet_test_cache.c

```diff
@@ -60,7 +60,10 @@ EFL_START_TEST(eet_test_cache_concurrency)
    Eina_Bool r;
    int tmpfd;
 
-   file = strdup("/tmp/eet_suite_testXXXXXX");
+   int path_size = strlen(eina_environment_tmp_get()) + strlen("eet_suite_testXXXXXX") + 2;
```

Comment:
Why +2?

Reply:
it uses `\\` to join paths

Reply:
oh, but it won't use `\\` on every system....is there an workarround?

Reply:
Did you mean something like `EINA_PATH_SEP`?

Reply:
> 
> 
> > Why +2?
> 
> path separator char + end of string char?
> 
> > oh, but it won't use `\\` on every system....is there an workarround?
> 
> Did you mean something like `EINA_PATH_SEP_C`?

`EINA_PATH_SEP_C` seems fair.

Also, htinking now: `\\` is not 2 characters, but 1. So the length should be increased by one. What if we saw what was the string before and after the join?

Reply:
> Did you mean something like EINA_PATH_SEP_C?

yeah, will try with `+ strlen(EINA_PATH_SEP_C)`, thanks @joaoantoniocardoso.

> Also, htinking now: \\ is not 2 characters, but 1. So the length should be increased by one. What if we saw what was the string before and after the join?

I will look at it, but if i remember it right I've tried it at the begining. Anyway, I will try.

Reply:
Thinking again: `EINA_PATH_SEP_C` is a `char` (while `EINA_PATH_SEP` is a `char*`/string, but it has the length of 1 character anyway), so the length would still be 1.

Reply:
oh sure, sure, `EINA_PATH_SEP_C` is a `char` and `EINA_PATH_SEP_S` is a `string` right, right.

About the sizes, I've run this on linux and windows:
```
   int path_size = strlen(eina_environment_tmp_get()) + strlen("eet_suite_testXXXXXX") + 2;
   file = malloc(sizeof(char)*path_size);
   eina_file_path_join(file, path_size
      , eina_environment_tmp_get(), "eet_suite_testXXXXXX");
fprintf(stderr, "+2 strlen(%s) = %llu\n", file, strlen(file));fflush(stderr);

free(file);
path_size = strlen(eina_environment_tmp_get()) + strlen("eet_suite_testXXXXXX") + 1;
file = malloc(sizeof(char)*path_size);
eina_file_path_join(file, path_size
  , eina_environment_tmp_get(), "eet_suite_testXXXXXX");
fprintf(stderr, "+1 strlen(%s) = %llu\n", file, strlen(file));fflush(stderr);
```
And in both cases we need the +2 to print all the path, we can see that with `+1` one of the `X` is cropped out:
```
+2 strlen(C:\Users\coku\AppData\Local\Temp\eet_suite_testXXXXXX) = 53
+1 strlen(C:\Users\coku\AppData\Local\Temp\eet_suite_testXXXXX) = 52
```
I'm pretty sure that @joaoantoniocardoso was right:
>path separator char + end of string char?

Reply:
Hmm just to confirm: what was the value of `eina_environment_tmp_get()`?

Reply:
Also, I don't believe it is an end of string, see:

> ```
> +2 strlen(C:\Users\coku\AppData\Local\Temp\eet_suite_testXXXXXX) = 53
> +1 strlen(C:\Users\coku\AppData\Local\Temp\eet_suite_testXXXXX) = 52
> ```

If it was a matter of end-of-string char, it would actually happend that +1 would print _more_ stuff, not less, since it wouldn't copy the end of string character. Since it has shown less, that means the last `X` is the char at `strlen-1` index, and not a `\0`.

Reply:
_\*sigh*_ forget what I said: I was thinking about the _displayed_ string length, but not about what the join would do internally.

Give a thumbs up to @Coquinho.

Reply:
hmm, path_size need to accommodate eina_environment_tmp_get() + EINA_PATH_SEP_S + "eet_suite_testXXXXXX" + "\0", so what is wrong with `strlen(eina_environment_tmp_get()) + strlen(EINA_PATH_SEP_S) + strlen("eet_suite_testXXXXXX") + 1`, or just `strlen(eina_environment_tmp_get()) + strlen("eet_suite_testXXXXXX") + 2`?

Reply:
Nothing, I was understanding the wrong thing here.